tests/test_project.py

```diff
@@ -403,3 +406,117 @@ def test_project_load_install(snippetcompiler_clean, install: bool) -> None:
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml;{raw_icon}"
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml;base64;{raw_icon}"
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml,base64;{raw_icon}"
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml,base64,{raw_icon}"
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xmlbase64{raw_icon}"
+    )
+    assert result.code == 400
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/svg+xml;base64{raw_icon}"
+    )
+    assert result.code == 400
+
+    # Check too large icon, base64 encoding increases the size
+    large_icon = base64.b64encode(b"a" * 65535).decode("utf-8")
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon=f"image/png;base64,{large_icon}"
+    )
+    assert result.code == 400
+
+    result = await client_v2.environment_create(
+        project_id=project_id_a, name="envx", description=desc, icon="image/png;base64,"
+    )
+    assert result.code == 400
+
+    # Test modification of the description and icon
+    name_of_env_to_modify = "env_no_icon"
+    result = await client_v2.environment_create(project_id=project_id_a, name=name_of_env_to_modify, description=desc, icon="")
+    assert result.code == 200
+    id_of_env_to_modify = result.result["data"]["id"]
+
+    icon_data_url = f"image/png;base64,{environment_icons['png']}"
+    # Add an icon
+    result = await client_v2.environment_modify(id_of_env_to_modify, name_of_env_to_modify, icon=icon_data_url)
+    assert result.code == 200
+    assert result.result["data"]["icon"] == icon_data_url
+
+    # Change the description, but keep the icon the same
+    result = await client_v2.environment_modify(id_of_env_to_modify, name_of_env_to_modify, description="new desc")
+    assert result.code == 200
+    assert result.result["data"]["icon"] == icon_data_url
+
+    # Delete the icon
+    result = await client_v2.environment_modify(id_of_env_to_modify, name_of_env_to_modify, icon="")
+    assert result.code == 200
+    assert result.result["data"]["icon"] == ""
+
+    result = await client_v2.environment_modify(id_of_env_to_modify, name_of_env_to_modify, description="b" * 256)
+    assert result.code == 400
```

Comment:
I would also add at least one test where the `environment_get` endpoint is used to verify that the change was applied in the database. As this an entirely different code path.